lib/improver/blending/weighted_blend.py

```diff
@@ -466,6 +595,15 @@ def process(self, cube, weights=None):
             msg = ('Coordinate to be collapsed not found in cube.')
             raise CoordinateNotFoundError(msg)
 
+        coord_dim = cube.coord_dims(self.coord)
+        if not coord_dim:
+            raise ValueError(
+                '{} has no associated dimension'.format(self.coord))
+
+        # Ensure input cube is ascending along blending coordinate, so that
+        # weights match cube slices
+        cube = sort_coord_in_cube(cube, self.coord, order="ascending")
```

Comment:
I'm not sure I understand why the sorting is being done

Reply:
I had to think about this because we are also doing a lot of sorting in the CLI, so it looks like we are doing this twice.  However, I wanted to cover this here as in theory, the plugin can be imported and used outside of the CLI (and may well be if this code is used outside of the UK).  Sorting here means we can say in the plugin docstring that the weights cube will always be returned in order of increasing value of the blend coordinate.

Reply:
Ok, and I've seen you've updated this docstring to say the weights correspond to the order of the points in the blending coord so this is ok.